README.md

```diff
@@ -34,3 +34,110 @@ Triggers, and any other element that can be used with `tektoncd/pipeline`).
   - Possibly driving API "feature"/changes
     Because we will write a lot of task, use them, … we should be able to find gap or
     enhancements in the API, and propose them as TEPs (with data).
+
+# Usage Examples
+
+
+This section explains how to use the tasks supported in this repository with the help of various tools like Tekton Resolvers as well as Pipelines as Code. 
+
+## Using Tekton Resolvers
+
+Make sure kubectl is installed, if not install it using this [link](https://kubernetes.io/docs/tasks/tools/).
+
+To use our tasks, you can create a Pipeline Resource as follows:
+
+```yaml
+---
+apiVersion: tekton.dev/v1beta1
+kind: Pipeline
+metadata:
+  labels:
+    name: example-pipeline
+  name: example-pipeline
+spec:
+  params:
+    # Customize the Params as needed by your chosen tasks
+    - name: APP_NAME
+      type: string
+      default: example
+    - name: IMAGE_PREFIX
+      type: string
+      default: "test"
+
+  workspaces:
+    - name: source
+
+  tasks:
+    # Add or Reference other tasks as needed
+    - name: example-task
+      taskRef:
+        resolver: git
+        params:
+          - name: url
+            value: https://github.com/openshift-pipelines/tektoncd-catalog.git
+          - name: revision
+            value: p
+          - name: pathInRepo
+            value: experimental/tasks/go-crane-image/v0.1.0/go-crane-image.yaml
+      workspaces:
+        - name: source
+          workspace: source
+      params:
+        - name: app
+          value: $(params.APP_NAME)
+        - name: image
+          value:
+            prefix: $(params.IMAGE_PREFIX)
+```
+Note: Add either a secret or update the service account with the git credentials
+
+Also for this example we have used a PersistentVolumeClaim as follows:
+
+```yaml
+---
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  labels:
+    name: test
+  name: test
+spec:
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 250Mi
+```
+
+Then use the following commands to apply & run the above Pipeline
+
+- Create PVC resource: kubectl apply -f pvc.yaml
+- Create Pipeline: kubectl apply -f pipeline.yaml
+- PipelineRun: tkn pipeline start example-pipeline --workspace="name=source,claimName=test,subPath=source" --showlog
```

Comment:
Let's have each command to be executed as a code block, please. Easier to recognize as a shell command and to copy-and-paste.